zavod/zavod/helpers/html.py

```diff
@@ -129,3 +129,37 @@ def links_to_dict(el: Element) -> Dict[str | None, str | None]:
     return {
         slugify(element_text(a), sep="_"): a.get("href") for a in el.findall(".//a")
     }
+
+
+def xpath_elements(
+    el: Element, xpath: str, *, expect_exactly: Optional[int] = None
+) -> List[Element]:
+    """
+    Return a list of HtmlElement objects that match the given XPath expression.
+    """
+    result = el.xpath(xpath)
+    if not isinstance(result, list) or not all(isinstance(r, Element) for r in result):
+        raise ValueError(
+            f"Expected list[Element] as result of xpath, got {type(result)}"
+        )
+    if expect_exactly is not None and len(result) != expect_exactly:
+        raise ValueError(
+            f"Expected {expect_exactly} elements, got {len(result)} for xpath {xpath!r}"
+        )
+    return [cast(Element, r) for r in result]
+
+
+def xpath_strings(
+    el: Element, xpath: str, *, expect_exactly: Optional[int] = None
+) -> List[str]:
+    """
+    Return a list of strings that match the given XPath expression.
+    """
+    result = el.xpath(xpath)
+    if not isinstance(result, list) or not all(isinstance(r, str) for r in result):
+        raise ValueError(f"Expected list[str] as result of xpath, got {type(result)}")
+    if expect_exactly is not None and len(result) != expect_exactly:
+        raise ValueError(
+            f"Expected {expect_exactly} elements, got {len(result)} for xpath {xpath!r}"
+        )
+    return [cast(str, r) for r in result]
```

Comment:
HTML strings often contain weird characters from HTML entities, like `\x0a`. Maybe we can filter using `normality.cleaning:remove_unsafe_chars`